### DISCUSSION TITLE
Electronics noob: Are capacitors and resistors needed when using Adafruit RP2040 Scorpio + HD107S/SK6812?

### DISCUSSION THREAD

**Discussion by markgabrang** [awawa-dev/HyperSerialPico#33] [Q&A] [answered]
Hi again @awawa-dev!

I see different diagrams where some people add resistors to the data lines/capacitors on the power line:
https://cdn-learn.adafruit.com/assets/assets/000/030/892/original/leds_Wiring-Diagram.png?1456961114

Are those required? I see some diagrams without resistors and capacitors.

I'm using the Adafruit RP2040 Scorpio + HD107S/SK6812 + Meanwell LRS-200-5.

Edit:
I'll be using about 180 LEDs on the HD107s and 144 LEDs on the SK6812 on separate RP2040 Scorpio.
The two strips will use the same Meanwell LRS-200-5 PSU.

Again, thank you for what you do! 🙏

**Comment by awawa-dev** [comment, marked as answer]
Hi
The capacitor eliminates some voltage fluctuations and the resistor limits the current, protecting the data lines and the LEDs themselves (sometimes LED manufacturers place it before 1 LED on the strip). I don't use them in my setups, but I don't want to suggest anything.

**Reply by markgabrang** [reply]
I might go with just the capacitor, since I'll only be using 10cm data lines:
https://quinled.info/data-signal-cable-conditioning/

Thanks again!
